tests/manage/storageclass/test_create_storageclass_with_same_name.py

```diff
@@ -51,9 +52,27 @@ def create_storageclass(sc_name, expect_fail=False):
     """
 
     # Create a storage class
-    sc_data = templating.load_yaml_to_dict(constants.CSI_RBD_STORAGECLASS_YAML)
+    rook_example_sc_data = templating.load_yaml_to_dict(
+        constants.ROOK_CSI_RBD_STORAGECLASS_YAML, multi_document=True
+    )
+    sc_data = templating.get_n_document_from_yaml(rook_example_sc_data, 1)
+    namespace = config.ENV_DATA['cluster_namespace']
+    sc_data['provisioner'] = f'{namespace}.rbd.csi.ceph.com'
     sc_data['metadata']['name'] = sc_name
-    sc_data['parameters']['clusterID'] = defaults.ROOK_CLUSTER_NAMESPACE
+    sc_data['parameters']['clusterID'] = namespace
+    sc_data['metadata']['pool'] = 'rbd'
+    sc_data['parameters'][
```

Comment:
why is below assignment needed? doesn't it come from dict?

Reply:
If I compared what we currently have in our template it's differ from what is currently in the template from rook. You can see diff here:
https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/templates/CSI/rbd/storageclass.yaml#L22 . we have csi-rbd-secret . 

But in rook example they have:
https://github.com/rook/rook/blob/master/cluster/examples/kubernetes/ceph/csi/rbd/storageclass.yaml#L32
rook-ceph-csi .

As I don't want to break anything by this PR and just want to make sure that everything is as before we need to have this change here to keep it as it was.

Reply:
I think since the files are coming from rook repo, we should use them as is and if they fail its a bug?

Reply:
Not really, we cannot rely only on the files which are on examples. We have more complex tests which needs to change a lot of parameters there starting with the namespace as in rook repo they use rook-ceph and we use openshift-storage to more complex parameters which needs to be set for test needs.  You can see the example of this helper method: https://github.com/red-hat-storage/ocs-ci/blob/master/tests/helpers.py#L239 . This deal with a lot of parameters.

Reply:
Lets take one assignment example: 
```
  sc_data['parameters'][
        'csi.storage.k8s.io/provisioner-secret-namespace'
    ] = 'default'
```
Why is this needed? we are not assigning any dynamic value here just a static 'default' ,  Upstream repo should already have static values assigned properly?

Reply:
As I wrote @vasukulkarni I am not author of this test and didn't want to change any behavior here. As I just compared the files already mentioned in comment above and changed values accordingly. 

>If I compared what we currently have in our template it's differ from what is currently in the template from rook. You can see diff here:
>https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/templates/CSI/rbd/storageclass.yaml#L22 . we have csi-rbd-secret .
>
>But in rook example they have:
>https://github.com/rook/rook/blob/master/cluster/examples/kubernetes/ceph/csi/rbd/storageclass.yaml#L32
rook-ceph-csi .

Just take a look in those files and differences. I don't really know what is the logic behind this but `csi.storage.k8s.io/provisioner-secret-namespace: rook-ceph` doesn't sounds good to me as we don't use `rook-ceph` namespace at all.

Reply:
@petr-balogh I think we can file an issue for this one and the owner of the test case can take care of it later,  It does sound like a bug if we are deviating from the standard templates.

Reply:
The problem is that the examples in rook repository are not official templates, those are just examples of the manifests and not exact files which needs or should be used.  User can change parameters, add new or do whatever they need, they just have provided files.  There are also real templates in examples here:  https://github.com/rook/rook/tree/master/cluster/examples/kubernetes/ceph/csi/template

which are real openshift templates, for those files I agree there shouldn't be any changes and should be created like they are and param should be passed through OC.